hw/xbox/mcpx_apu.c

```diff
@@ -826,12 +845,38 @@ static void se_frame(void *opaque)
                     NV_PAVS_VOICE_PAR_STATE_ACTIVE_VOICE)) {
                 MCPX_DPRINTF("voice %d not active...!\n", d->regs[current]);
                 fe_method(d, SE2FE_IDLE_VOICE, d->regs[current]);
+            } else {
+                process_voice(d, mixbin, d->regs[current]);
             }
             MCPX_DPRINTF("next voice %d\n", d->regs[next]);
             d->regs[current] = d->regs[next];
         }
     }
 
+#if INJECT_VP_BEEP
+    /* Inject some audio to the mixbin for debugging.
+     * Signal is 1500 Hz sine wave, phase shifted by mixbin number.*/
+    for (unsigned int i = 0; i < NUM_MIXBINS; i++) {
+        for (unsigned int j = 0; j < NUM_SAMPLES_PER_FRAME; j++) {
+            /* Avoid multiple of 1.0 / NUM_SAMPLES_PER_FRAME for phase shift,
+             * or waves cancel out */
+            float offset = j / (float)NUM_SAMPLES_PER_FRAME -
+                           i / (float)(NUM_SAMPLES_PER_FRAME + 1);
+            float wave = sinf(offset * M_PI * 2.0f);
+            mixbin[i][j] += wave * 0x3FFFFF;
```

Comment:
This used to be 0x7FFFFF during testing (which was way too loud and sounded like a fire alarm).
I have not tested if this new volume is high enough - I will do that soon.